heron/shell/src/python/handlers/killexecutorhandler.py

```diff
@@ -28,16 +28,38 @@ class KillExecutorHandler(tornado.web.RequestHandler):
   @tornado.web.asynchronous
   def post(self):
     """ post method """
+    def status_finish(ret):
+      self.set_status(ret)
+      self.finish()
+
+    def kill_parent():
+      status_finish(200)
+      logger.info("Killing parent executor")
+      os.killpg(os.getppid(), signal.SIGTERM)
+
     logger = logging.getLogger(__file__)
     logger.info("Received 'Killing parent executor' request")
     data = dict(urlparse.parse_qsl(self.request.body))
+
+    # check shared secret
     sharedSecret = data.get('secret')
     if sharedSecret != options.secret:
-      self.set_status(403)
-      self.finish()
+      status_finish(403)
       return
-    logger.info("Killing parent executor response 200")
-    self.set_status(200)
-    self.finish()
-    logger.info("Killing parent executor")
-    os.killpg(os.getppid(), signal.SIGTERM)
+
+    processName = data.get('process')
```

Comment:
The variable name `processName` is a bit confusing. It seems what you refer to is `instanceId`, rather than the `processName`. Similarly, I would recommend changing the `request` structure so we can have `instance_id_to_restart` here

Reply:
done